### PR TITLE
Fix deprecation warning missfires

### LINKED ISSUE #1476: `universe.segment` and `universe.residues` triger a deprecation warning
Likeky following #1403, calling the `segments` property of a universe issues the following deprecation warning:

```
/home/jon/dev/mdanalysis/package/MDAnalysis/core/topologyattrs.py:1268: DeprecationWarning: `_get_named_segment` is deprecated!
Instant selector SegmentGroup.<name> is deprecated and will be removed in 1.0. Use SegmentGroup[SegmentGroup.segids == '<name>'] instead.
  return segmentgroup._get_named_segment(segid)
```

The `residues` property issues the following one:

```
/home/jon/dev/mdanalysis/package/MDAnalysis/core/topologyattrs.py:1064: DeprecationWarning: `_get_named_residue` is deprecated!
Instant selector ResidueGroup.<name> or Segment.<name> is deprecated and will be removed in 1.0. Use ResidueGroup[ResidueGroup.resnames == '<name>'] or Segment.residues[Segment.residues == '<name>'] instead.
  return residuegroup._get_named_residue(resname)
```

No warning should be emitted.

The `atoms` property does not emit warning as expected.

`Resnames._get_named_residue` and `Segments._get_named_segment` should handle the deprecation warning the same way as `Atoms._get_named_atom`; issue the warning only if the methods return something. See https://github.com/MDAnalysis/mdanalysis/pull/1403#issuecomment-309768319 and a9f9655a3b45b8486a1aadb5ad903f8bfe68ef3f.

### PR DESCRIPTION
**Pull request by jbarnoud**
`u.segments` and `u.residues` should not issue a deprecation warning.
This commit applies the same strategy than
a9f9655a3b45b8486a1aadb5ad903f8bfe68ef3f to avoid the unwanted warning.

Fixes  #1476

PR Checklist
------------
 - [x] Tests?
 - ~[ ] Docs?~
 - [x] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

**Comment by jbarnoud** [comment]
This still requires an entry in the CHANGELOG, and tests.

**Comment by jbarnoud** [comment]
Ready to go. Pytest is fun.

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py [hunk 1]
```diff
@@ -494,3 +495,45 @@ def test_set_crosslevel(self):
...
+        'universe.residues.LYS',
+        'universe.segments.s4AKE',
+        pytest.param('universe.s4AKE',
+                     marks=pytest.mark.xfail(reason="Issue #1478")),
+    ))
+    def test_deprecation(self, universe, instruction):
+        """Test that the warnings are issued when required.
+        """
+        with pytest.deprecated_call():
+            exec(instruction)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_topologyattrs.py]: so our linter doesn't like `exec`, you could change this to `getattr(universe, instruction)` where instruction is now just `atoms.CA` etc

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/core/test_topologyattrs.py]: I was going to add a comment for pylint to ignore the `exec` here. Using `exec` in this specific context makes both the test, and the verbose output of the test more readable. The warning about `exec` comes from the security risk that exists when `exec` is used on untrusted input; here the input is safe and cannot be forged to take advantage of the code.

Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_topologyattrs.py]: Yeah ok that's fine too, just fixup the linter so it doesn't cry about this then

**Pull request merged**